haystack/pipelines/base.py

```diff
@@ -717,7 +718,16 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
             component_type = component_instance.pipeline_config["type"]
             component_params = component_instance.pipeline_config["params"]
             components[node] = {"name": node, "type": component_type, "params": {}}
-            component_signature = inspect.signature(type(component_instance)).parameters
+            if type(component_instance) in [OpenSearchDocumentStore, OpenDistroElasticsearchDocumentStore]:
```

Comment:
Maybe we could make it a bit more generic here by using:
`if issubclass(type(component_instance), ElasticsearchDocumentStore)`?

Reply:
note that we can then also delete two of the imports: `import OpenSearchDocumentStore, OpenDistroElasticsearchDocumentStore,`

Reply:
Hi julian! Thanks for your input! After some research I found out that not only the Opensearch/OpenDistro documentstores extend classes with init parameters. So the original idea to fix this only for the Opensearch/Opendistro only fixes this bug for this specific case. My current approach unifies all parameters that exist for all parent classes. This way it should work for other classes as well. (e.g. milvus which is extended by sql).